specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -5448,88 +5645,766 @@
+                    "value": "Contains"
+                  },
+                  {
+                    "description": "Evaluates if the property does not contain any of the condition values",
+                    "value": "NotContains"
+                  },
+                  {
+                    "description": "Evaluates if the property starts with any of the condition values",
+                    "value": "StartsWith"
+                  },
+                  {
+                    "description": "Evaluates if the property does not start with any of the condition values",
+                    "value": "NotStartsWith"
+                  },
+                  {
+                    "description": "Evaluates if the property ends with any of the condition values",
+                    "value": "EndsWith"
+                  },
+                  {
+                    "description": "Evaluates if the property does not end with any of the condition values",
+                    "value": "NotEndsWith"
+                  }
+                ]
+              }
+            },
+            "propertyValues": {
+              "description": "The values to use for evaluating the condition",
+              "items": {
+                "description": "A value to use for evaluating the condition",
+                "type": "string"
+              },
+              "type": "array"
+            }
+          },
+          "type": "object"
+        }
+      },
+      "required": [
+        "conditionProperties"
+      ]
+    },
+    "AutomationRulesList": {
+      "description": "List all the automation rules.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of automation rules.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of automation rules.",
+          "items": {
+            "$ref": "#/definitions/AutomationRule"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "AutomationRuleProperties": {
+      "description": "Describes automation rule properties",
+      "properties": {
+        "displayName": {
+          "description": "The display name of the automation  rule",
+          "type": "string"
+        },
+        "order": {
+          "description": "The order of execution of the automation rule",
+          "type": "integer",
+          "format": "int32"
+        },
+        "triggeringLogic": {
+          "$ref": "#/definitions/AutomationRuleTriggeringLogic",
+          "description": "The triggering logic of the automation rule",
+          "type": "object"
+        },
+        "actions": {
+          "description": "The actions to execute when the automation rule is triggered",
+          "items": {
+            "$ref": "#/definitions/AutomationRuleAction"
+          },
+          "type": "array"
+        },
+        "createdTimeUtc": {
```

Comment:
Isn't this redundant given that starting from the next version we will need to add the systemData property?

Reply:
we currently have to use the 2019-01-01-preview version which doesn't have systemData in the ARM resource contract since the preview customers already use that version and rely on it. when we add automation rules to the next API version we may decide to change the response contract to rely only on systemData for the resource metadata (date/caller of create/update), though I'm still not sure we'll want to since systemData shows exposes very limited information on the user that performed the call that may not be sufficient for OData filtering capabilities we'll want to add it the future